app/lib/shared/urls.dart

```diff
@@ -154,30 +154,32 @@ String pkgDocUrl(
   String? version,
   bool includeHost = false,
   String? relativePath,
-  bool omitTrailingSlash = false,
   bool isLatest = false,
 }) {
+  final parsedRelativePathSegments = relativePath == null
+      ? const <String>[]
+      : Uri.parse(relativePath)
+          .pathSegments
+          .where((e) => e.isNotEmpty)
+          .toList();
+
   if (isLatest || version == null) {
     version = 'latest';
   }
   final segments = <String>[
     'documentation',
     package,
     version,
+    ...parsedRelativePathSegments,
   ];
   final baseUri = includeHost ? _siteRootUri : _pathRootUri;
   String url = baseUri.resolveUri(Uri(pathSegments: segments)).toString();
 
-  if (relativePath != null) {
-    url = p.join(url, relativePath);
-  } else if (!omitTrailingSlash) {
-    url = '$url/';
-  }
   if (url.endsWith('/index.html')) {
```

Comment:
Could this be done better by testing the last segment?

Reply:
It could work with something like this:
```
    ...parsedRelativePathSegments
        // filter out the last segment if it is `index.html` 
        .whereIndexed((index, element) =>
            index != parsedRelativePathSegments.length - 1 ||
            element != 'index.html'),
```

However, I don't find it easier to read. Maybe `if (segments.isNotEmpty && segments.last == 'index.html`) segments.removeLast();` could work, but then it will make the trailing slash check a bit more complex. Which would you prefer?

Reply:
I was thinking something like segments.removeLast();

Also the trailing-slash check could probably be done by checking if the last segment is empty.

But I'm not sure I know the perfect solution. Just all these string-based uri operations make me a bit worried.

Reply:
Updated, PTAL.